src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/scalable/TruthSensitivityTranche.java

```diff
@@ -0,0 +1,386 @@
+
+                final String[] vals = line.split(VALUE_SEPARATOR);
+                if (header == null) {  //reading the header
+                    header = vals;
+                    if (header.length != EXPECTED_COLUMN_COUNT) {
+                        throw new UserException.MalformedFile(f, String.format("Expected %d elements in header line %s", EXPECTED_COLUMN_COUNT, line));
+                    }
+                } else {
+                    if (header.length != vals.length) {
+                        throw new UserException.MalformedFile(f, "Line had too few/many fields.  Header = " + header.length + " vals " + vals.length + ". The line was: " + line);
+                    }
+
+                    Map<String, String> bindings = new LinkedHashMap<>();
+                    for (int i = 0; i < vals.length; i++) {
+                        bindings.put(header[i], vals[i]);
+                    }
+                    tranches.add(new TruthSensitivityTranche(
+                            getRequiredDouble(bindings, "targetTruthSensitivity"),
+                            getRequiredInteger(bindings, "numSites"),
+                            getRequiredDouble(bindings, "tiTv"),
+                            getRequiredDouble(bindings, "minScore"),
+                            bindings.get("filterName"),
+                            VariantTypeMode.valueOf(bindings.get("mode")),
+                            getOptionalInteger(bindings, "accessibleTruthSites", -1),
+                            getOptionalInteger(bindings, "callsAtTruthSites", -1)
+                    ));
+                }
+            }
+        }
+
+        tranches.sort(TRUTH_SENSITIVITY_ORDER);
+        return tranches;
+    }
+
+    @VisibleForTesting
+    static final class TruthSensitivityMetric {
+        private final String name;
+        private double[] runningSensitivity;
+        private final int nTrueSites;
+
+        TruthSensitivityMetric(final int nTrueSites) {
+            this.name = "TruthSensitivity";
+            this.nTrueSites = nTrueSites;
+        }
+
+        public String getName(){
+            return name;
+        }
+
+        double getRunningMetric(final int i) {
+            return runningSensitivity[i];
+        }
+
+        private static double getThreshold(final double tranche) {
+            return 1. - tranche / 100.; // tranche of 1 => 99% sensitivity target
+        }
+
+        private void calculateRunningMetric(final List<Boolean> isTruth) {
+            int nCalledAtTruth = 0;
+            runningSensitivity = new double[isTruth.size()];
+
+            for (int i = isTruth.size() - 1; i >= 0; i--) {
+                nCalledAtTruth += isTruth.get(i) ? 1 : 0;
+                runningSensitivity[i] = 1 - nCalledAtTruth / (1. * nTrueSites);
+            }
+        }
+    }
+
+    // TODO clean all this up once VariantDataManager is refactored
+    static List<TruthSensitivityTranche> findTranches(final List<Double> scores,
```

Comment:
I feel like the fundamental job here, figuring out different score thresholds, which depends only on a scoring function and truth labels, should be decoupled from particular metrics like Ti/Tv.  That is, first calculate the thresholds, then calculate the QC metrics, without necessarily storing them in the thresholds class.  As long as the scores are stored the price is just an extra O(N) pass over the data, but then again you save the O(N log(N)) sorts below.

Reply:
I think I got rid of some classes and interaction with DataManager, which necessitate the additional sorts below for now. Definitely should clean up! But all of these sorts are quite cheap.